src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrUpdater.cs

```diff
@@ -103,12 +104,30 @@ public class VmrUpdater : VmrManagerBase, IVmrUpdater
         bool updateDependencies,
         CancellationToken cancellationToken)
     {
-        await _dependencyTracker.InitializeSourceMappings(_vmrInfo.VmrPath / VmrInfo.SourcesDir / VmrInfo.SourceMappingsFileName);
-        
+        await _dependencyTracker
+            .InitializeSourceMappings(_vmrInfo.VmrPath / VmrInfo.SourcesDir / VmrInfo.SourceMappingsFileName);
+
         var mapping = _dependencyTracker.Mappings
             .FirstOrDefault(m => m.Name.Equals(mappingName, StringComparison.InvariantCultureIgnoreCase))
             ?? throw new Exception($"No mapping named '{mappingName}' found");
 
+        if (_vmrInfo.SourceMappingsPath != null
+            && _vmrInfo.SourceMappingsPath.StartsWith(VmrInfo.GetRelativeRepoSourcesPath(mapping)))
+        {
+            var fileRelativePath = _vmrInfo.SourceMappingsPath.Substring(VmrInfo.GetRelativeRepoSourcesPath(mapping).Length);
+            var clonePath = await _cloneManager.PrepareClone(
+                mapping.DefaultRemote, 
+                targetRevision ?? mapping.DefaultRef, 
+                cancellationToken);
+            
+            _logger.LogDebug($"Loading a new version of source mappings from {clonePath / fileRelativePath}");
+            await _dependencyTracker.InitializeSourceMappings(clonePath / fileRelativePath);
```

Comment:
Can we not initialize twice?

Reply:
> Can we not initialize twice?

we need the remote for the clone

Reply:
We should use SourceManifest for that actually

Reply:
Hm, on another thought. I think we should do something else.. I will explain this tomorrow

Reply:
@MilenaHristova let's merge it as-is and I can make additional changes as part of https://github.com/dotnet/arcade/issues/11386